lib/plugins/inventory.js

```diff
@@ -418,7 +423,7 @@ function inject (bot, { hideErrors }) {
       }
     } else if (window.type === 'minecraft:merchant') {
       const toUpdate = []
-      if (slot <= 2 && !window.selectedTrade.tradeDisabled && expectTradeUpdate(window)) {
```

Comment:
The `updateSlot:2` event is already awaited by `promisePutAway` in `putAway()`, so if we await it here too, we'll never get it. I think `waitForWindowUpdate` should only wait for updates on slots other than the slot it was called on.